pkg/runtime/framework/plugins/torch/torch_nproc_per_node_test.go

```diff
@@ -0,0 +1,265 @@
+/*
+Copyright 2024 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package torch contains tests for the Torch plugin, including validation of the fix for Issue #2407
+// which caps nproc_per_node based on CPU resources when set to "auto" and no GPU is requested.
+package torch
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	trainer "github.com/kubeflow/trainer/pkg/apis/trainer/v1alpha1"
+	"github.com/kubeflow/trainer/pkg/constants"
+	"github.com/kubeflow/trainer/pkg/runtime"
+	utiltesting "github.com/kubeflow/trainer/pkg/util/testing"
+)
+
+func TestTorch_EnforceMLPolicy_NumProcPerNode_Issue2407(t *testing.T) {
```

Comment:
I think, these test cases should be part of TestTorch similar to what we've done for MPI here: https://github.com/kubeflow/trainer/pull/2481